### PR TITLE
Jackson version upgrade: default from 2.4.2 to 2.7.4 (latest stable)

### PR DESCRIPTION
**Pull request by cowtowncoder**
Fix for [FLINK-3855] -- upgrade default Jackson version from 2.4.2 to 2.5.5 (last 2.5 patch); remove unnecessary explicit version ref, and upgrade ES client to latest patch for minor version it refers.

### CODE REVIEW

File: flink-streaming-connectors/flink-connector-elasticsearch2/pom.xml [hunk 1]
```diff
@@ -55,11 +55,11 @@ under the License.
...
 
-		<dependency>
-			<groupId>com.fasterxml.jackson.core</groupId>
-			<artifactId>jackson-core</artifactId>
-			<version>2.7.2</version>
-		</dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-core</artifactId>
+            <version>2.7.4</version>
```
Comment (smarthi) [inline flink-streaming-connectors/flink-connector-elasticsearch2/pom.xml]: Replace with ${jackson.version} if the parent pom value for this is being changed to 2.7.2

**Comment by aljoscha** [comment]
@fhueske jackson-databind is only used in the Table API to render the explain output. We should be good to update to latest there as well, right?

**Comment by fhueske** [comment]
Yes, I hope so.

**Comment by aljoscha** [comment]
@cowtowncoder do you see a reason for not upgrading all the way to 2.7?

**Comment by cowtowncoder** [comment]
@aljoscha @fhueske Nothing special, just thought I'd start with smallest step, given that this is my first contribution here.
But given that 2.7.4 is out now, I agree that going right there does make most sense and should be safe.
It also looks like most usage is via Tree API (JsonNodes), some streaming; most changes are typically in databinding and very few compatibility issues occur outside databinding. So to me upgrade seems safe either way. And since Elastic-client already uses 2.7 that would allow unification of versions as well.

**Comment by smarthi** [comment]
ElasticSearch 2.x requires jackson >= 2.6.2, hence the Elastic 2.x connector jackson version is set at 2.7.x.  Locally, I changed the Jackson version to be 2.7.x in the parent pom and didn't see any issues or tests failures. I think its safe to change the jackson.version to 2.7.4 in parent pom. If so, please remove the <jackson.version> in ElasticSearch2/pom.xml

**Comment by cowtowncoder** [comment]
@smarthi Updated as suggested.

**Comment by aljoscha** [comment]
The build now fails consistently for the wikipedia connector? Could you push another dummy commit to rerun travis tests. Let's see if it goes away.

**Comment by smarthi** [comment]
There IRC Wikimedia channel is timing out and hence the reason for Wiki test failures, nothing to do with this PR.

LGTM IMO

**Comment by aljoscha** [comment]
I'm running a last local test and then merging this. Thanks for the work! 😃

**Comment by fhueske** [comment]
Merging

**Pull request closed without merging**